### PR TITLE
JW-1149 updated vendor packages

### PR DESCRIPTION
**Pull request by mgreg0r**

### CODE REVIEW

**Comment by mgreg0r** [comment]
```
licenses.exe github.com/codilime/contrail-windows-docker
github.com/codilime/contrail-windows-docker                                                          ? ("Do What The F*ck You Want To Public License",  4%)
github.com/codilime/contrail-windows-docker/vendor/github.com/Juniper/contrail-go-api                Apache License 2.0
github.com/codilime/contrail-windows-docker/vendor/github.com/Microsoft/go-winio                     MIT License
github.com/codilime/contrail-windows-docker/vendor/github.com/Microsoft/hcsshim                      MIT License
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/distribution                    Apache License 2.0
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/docker                          Apache License 2.0 (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/go-connections                  Apache License 2.0 (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/go-plugins-helpers              Apache License 2.0
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/go-units                        Apache License 2.0 (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/docker/libnetwork/netlabel             Apache License 2.0
github.com/codilime/contrail-windows-docker/vendor/github.com/gogo/protobuf/proto                    BSD 3-clause "New" or "Revised" License (90%)
github.com/codilime/contrail-windows-docker/vendor/github.com/golang/glog                            Apache License 2.0
github.com/codilime/contrail-windows-docker/vendor/github.com/kardianos/osext                        BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/onsi/gomega                            MIT License (98%)
github.com/codilime/contrail-windows-docker/vendor/github.com/opencontainers/go-digest               Apache License 2.0 (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/opencontainers/image-spec/specs-go/v1  Apache License 2.0 (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/pborman/uuid                           BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/github.com/pkg/errors                             BSD 2-clause "Simplified" License
github.com/codilime/contrail-windows-docker/vendor/github.com/sirupsen/logrus                        MIT License
github.com/codilime/contrail-windows-docker/vendor/golang.org/x/crypto/ssh/terminal                  BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/golang.org/x/net                                  BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/golang.org/x/sys/windows/svc/debug                BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/golang.org/x/text                                 BSD 3-clause "New" or "Revised" License (96%)
github.com/codilime/contrail-windows-docker/vendor/gopkg.in/yaml.v2                                  Apache License 2.0
```

File: vendor/vendor.json [hunk 1]
```diff
@@ -987,76 +1107,210 @@
...
-			"checksumSHA1": "LJNj7E8xswl15uf17DQgwhVMhyA=",
+			"checksumSHA1": "t5W90J4d7vhrs6+LDp0BalvCZEU=",
+			"path": "github.com/sirupsen/logrus",
+			"revision": "89742aefa4b206dcf400792f3bd35b542998eb3b",
+			"revisionTime": "2017-08-22T13:27:46Z"
+		},
+		{
+			"path": "github.com/ugorji",
+			"revision": ""
+		},
```
Comment (m-kostrzewa) [inline vendor/vendor.json]: Why is this added if there is no revision specified?

File: vendor/vendor.json [hunk 2]
```diff
@@ -987,76 +1107,210 @@
 			"revisionTime": "2017-03-13T16:33:22Z"
 		},
 		{
-			"checksumSHA1": "LJNj7E8xswl15uf17DQgwhVMhyA=",
+			"checksumSHA1": "t5W90J4d7vhrs6+LDp0BalvCZEU=",
+			"path": "github.com/sirupsen/logrus",
+			"revision": "89742aefa4b206dcf400792f3bd35b542998eb3b",
+			"revisionTime": "2017-08-22T13:27:46Z"
+		},
```
Comment (m-kostrzewa) [inline vendor/vendor.json]: logrus is added but I don't see it removed anywhere (it was here before)?...

File: vendor/vendor.json [hunk 3]
```diff
@@ -987,76 +1107,210 @@
...
+			"checksumSHA1": "G9ZGa1NLkstgPd9EylK9Hq+wqrE=",
 			"path": "github.com/ugorji/go/codec",
-			"revision": "708a42d246822952f38190a8d8c4e6b16a0e600c",
-			"revisionTime": "2017-03-12T11:21:14Z"
+			"revision": "8c0409fcbb70099c748d71f714529204975f6c3f",
+			"revisionTime": "2017-08-26T15:59:43Z"
+		},
+		{
+			"path": "github.com/vishvanada/netlink",
+			"revision": ""
```
Comment (m-kostrzewa) [inline vendor/vendor.json]: This is added but `revision` is empty. Why?

**Comment by m-kostrzewa** [comment]
List of *added* packages:
```
+			"path": "github.com/coreos/etcd/pkg/srv",
+			"path": "github.com/docker/docker/api/types/swarm/runtime",
+			"path": "github.com/docker/docker/pkg/mount",
+			"path": "github.com/docker/libnetwork/common",
+			"path": "github.com/docker/libnetwork/diagnose",
+			"path": "github.com/hashicorp/go-immutable-radix",
+			"path": "github.com/hashicorp/go-rootcerts",
+			"path": "github.com/hashicorp/golang-lru/simplelru",
+			"path": "github.com/miekg/dns",
+			"path": "github.com/mitchellh/go-homedir",
+			"path": "github.com/onsi/ginkgo/ginkgo/convert",
+			"path": "github.com/onsi/ginkgo/ginkgo/interrupthandler",
+			"path": "github.com/onsi/ginkgo/ginkgo/nodot",
+			"path": "github.com/onsi/ginkgo/ginkgo/testrunner",
+			"path": "github.com/onsi/ginkgo/ginkgo/testsuite",
+			"path": "github.com/onsi/ginkgo/ginkgo/watch",
+			"path": "github.com/onsi/ginkgo/internal/spec_iterator",
+			"path": "github.com/opencontainers/image-spec/specs-go",
+			"path": "github.com/opencontainers/image-spec/specs-go/v1",
+			"path": "github.com/opencontainers/runc/libcontainer/configs",
+			"path": "github.com/sirupsen/logrus",
+			"path": "github.com/ugorji",
+			"path": "github.com/ugorji/go",
+			"path": "github.com/vishvanada/netlink",
+			"path": "golang.org/x/crypto/ssh/terminal",
+			"path": "golang.org/x/net/html",
+			"path": "golang.org/x/net/html/atom",
+			"path": "golang.org/x/net/html/charset",
+			"path": "golang.org/x/text/encoding",
+			"path": "golang.org/x/text/encoding/charmap",
+			"path": "golang.org/x/text/encoding/htmlindex",
+			"path": "golang.org/x/text/encoding/internal",
+			"path": "golang.org/x/text/encoding/internal/identifier",
+			"path": "golang.org/x/text/encoding/japanese",
+			"path": "golang.org/x/text/encoding/korean",
+			"path": "golang.org/x/text/encoding/simplifiedchinese",
+			"path": "golang.org/x/text/encoding/traditionalchinese",
+			"path": "golang.org/x/text/encoding/unicode",
+			"path": "golang.org/x/text/internal/tag",
+			"path": "golang.org/x/text/internal/utf8internal",
+			"path": "golang.org/x/text/language",
+			"path": "golang.org/x/text/runes",
+			"path": "golang.org/x/text/transform",
```

**Review by m-kostrzewa — COMMENTED** [review]
I've reviewed vendor.json and have some questions. It seems like some dependencies became obsolete but were not removed from vendor.json for some reason?

File: driver/driver_test.go [hunk 4]
```diff
@@ -767,7 +787,11 @@ var _ = Describe("On requests from docker daemon", func() {
 
 		BeforeEach(func() {
 			_, dockerNetID, containerID = setupNetworksAndEndpoints(contrailController, docker)
-			dockerNet, err := docker.NetworkInspect(context.Background(), dockerNetID, false)
+			inspectOptions := dockerTypes.NetworkInspectOptions{
+				Scope:	 "",
+				Verbose: false,
+			}
+			dockerNet, err := docker.NetworkInspect(context.Background(), dockerNetID, inspectOptions)
```
Comment (m-kostrzewa) [inline driver/driver_test.go]: Since this got much more verbose, I think we should wrap this whole 5-line block in some helper function

**Review by m-kostrzewa — CHANGES_REQUESTED** [review]
Looks good, but I think we should squash the first and third commits so that we don't add useless heavy history. We should leave the commit with fixes separately ofc.

**Review by m-kostrzewa — COMMENTED** [review]
@mgreg0r as I mentioned, I think we shouldn't "Squash and merge" this PR blindly - it should contain two commits: 1. update vendors, 2. fixes. But for that to work we would need to rebase and squash the first and third commits (JW-1149 updated vendor packages and JW-1149 removed unused packages) into one, and the second and fourths also into one. Then we just "Merge" those 2 commits into master without squashing them together. 

The reason is that if we squash "fixes" with "update vendor stuff" we won't be able to easily see what needed to be fixed since there's hundreds of thousands of lines of changes.

@jablonskim what do you think?

**Review by jablonskim — APPROVED** [review]

**Comment by mgreg0r** [comment]
Janie, przetestuj proszę

File: hns/hns.go [hunk 5]
```diff
@@ -169,7 +169,7 @@ func GetHNSEndpointByName(name string) (*hcsshim.HNSEndpoint, error) {
 }
 
 func ListHNSEndpoints() ([]hcsshim.HNSEndpoint, error) {
-	endpoints, err := hcsshim.HNSListEndpointRequest("GET", "", "")
+	endpoints, err := hcsshim.HNSListEndpointRequest()
```
Comment (m-kostrzewa) [inline hns/hns.go:172]: Totally had this feature implemented for them :'( https://github.com/Microsoft/hcsshim/pull/104

**Review by m-kostrzewa — APPROVED** [review]
Thx for rebase, lgtm

**Pull request merged**